src/layers/model.rs

```diff
@@ -0,0 +1,25 @@
+use indexmap::IndexMap;
+use serde_pickle::{de, ser};
+use std::fs::File;
+
+pub trait Model {
+    fn save(&self, path: &str) {
+        let mut file = File::create(path).unwrap();
+
+        let state_dict = self.state_dict();
+
+        ser::to_writer(&mut file, &state_dict, true).unwrap();
+    }
```

Comment:
@Demba3 you commented on this, how about naming it `save_state_dict`? Although it is different from Pytorch the method name explicitly denotes the functionality